Modules/_ssl.c

```diff
@@ -2996,6 +2996,125 @@ static PyType_Spec PySSLSocket_spec = {
     .slots = PySSLSocket_slots,
 };
 
+#ifdef _MSC_VER
+static const unsigned char *
+_get_cert_bytes(const X509 *cert, int *length)
+{
+    unsigned char *cert_bytes;
+    int cert_bytes_length = i2d_X509(cert, NULL);
+    if (cert_bytes_length <= 0) {
+        return NULL;
+    }
+
+    cert_bytes = PyMem_RawMalloc(cert_bytes_length);
+    if (cert_bytes == NULL) {
+        return NULL;
+    }
+
+    if (i2d_X509(cert, &cert_bytes) <= 0) {
+        return NULL;
+    }
+
+    *length = cert_bytes_length;
+
+    /* i2d_X509 moves the input pointer to the end of the data */
+    return cert_bytes - cert_bytes_length;
+}
+
+
+static int
+_translate_policy_status_error(int error)
+{
+    switch (error) {
+    case TRUST_E_CERT_SIGNATURE:
+        return X509_V_ERR_CERT_SIGNATURE_FAILURE;
+    case CERT_E_UNTRUSTEDROOT:
+        return X509_V_ERR_CERT_UNTRUSTED;
+    case CERT_E_CHAINING:
+        return X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT;
+    case CERT_E_EXPIRED:
+        return X509_V_ERR_CERT_HAS_EXPIRED;
+    case CERT_E_INVALID_POLICY:
+        return X509_V_ERR_INVALID_POLICY_EXTENSION;
```

Comment:
This patch seems to satisfy the other failing check in `test_ssl`, however, I'm not sure whether this is correct:

```suggestion
    case CERT_E_INVALID_POLICY:
        return X509_V_ERR_INVALID_CA;
```

Here are the relevant docs:

CERT_E_INVALID_POLICY:
The certificate has an invalid policy.

X509_V_ERR_INVALID_CA:
A CA certificate is invalid. Either it is not a CA or its extensions are
not consistent with the supplied purpose.


- https://docs.openssl.org/master/man3/X509_STORE_CTX_get_error/#error-codes
- https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-cert_chain_policy_status#members

Reply:
It looks like test_ssl isn't failing anymore actually, however, test_poplib is failing due to a different error